### PR TITLE
test(number input): add refactored tests for number input FE-4554

### PR DESCRIPTION
**Pull request by sianford**
### Proposed behaviour

Refactor for `number-input` component to use new cypress-component-react framework for testing.

### Current behaviour

Currently we are using the regular cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run npx cypress open-ct to check if there is newly added test.file
- [x] Check the number.test.js file passed
- [x] Check all tests have been moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file

### CODE REVIEW

File: src/components/number/number.test.js [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+  return <Number label="Number" value={state} onChange={setValue} {...props} />;
+};
+
+context("Tests for Number component", () => {
+  describe("check props for Number component", () => {
+    it.each(testData)(
+      "should render Number label using %s special characters",
+      (labelValue) => {
+        CypressMountWithProviders(<Number label={labelValue} />);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:36]:
```suggestion
        CypressMountWithProviders(<NumberInputComponent label={labelValue} />);
```

File: src/components/number/number.test.js [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+        getDataElementByValue("label").should("have.text", labelValue);
+      }
+    );
+
+    it.each(testData)(
+      "should render labelHelp message using %s special characters",
+      (labelHelpValue) => {
+        CypressMountWithProviders(
+          <Number label="Number" labelHelp={labelHelpValue} />
```
Comment (DlgSHi) [inline src/components/number/number.test.js:46]:
```suggestion
          <NumberInputComponent labelHelp={labelHelpValue} />
```

File: src/components/number/number.test.js [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+
+        helpIcon().trigger("mouseover");
+        tooltipPreview().should("have.text", labelHelpValue);
+      }
+    );
+
+    it.each(testData)(
+      "should render fieldHelp message using %s special characters",
+      (fieldHelpValue) => {
+        CypressMountWithProviders(<Number fieldHelp={fieldHelpValue} />);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:57]:
```suggestion
        CypressMountWithProviders(<NumberInputComponent fieldHelp={fieldHelpValue} />);
```

File: src/components/number/number.test.js [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+
+        fieldHelpPreview().should("have.text", fieldHelpValue);
+      }
+    );
+
+    it.each(testData)(
+      "should render prefix using %s special characters",
+      (prefixValue) => {
+        CypressMountWithProviders(<Number prefix={prefixValue} />);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:66]:
```suggestion
        CypressMountWithProviders(<NumberInputComponent prefix={prefixValue} />);
```

File: src/components/number/number.test.js [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+
+        commonInputPrefix().should("have.text", prefixValue);
+      }
+    );
+
+    it.each(testData)(
+      "should render placeholder using %s special characters",
+      (placeholderValue) => {
+        CypressMountWithProviders(<Number placeholder={placeholderValue} />);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:75]:
```suggestion
        CypressMountWithProviders(<NumberInputComponent placeholder={placeholderValue} />);
```

File: src/components/number/number.test.js [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+          .then(() => {
+            commonInputCharacterLimit()
+              .should("have.text", `${charactersUsed}/${limit}`)
+              .and("have.css", "color", color);
+          });
+      }
+    );
+
+    it("should check add icon inside of the Number renders", () => {
+      CypressMountWithProviders(<Number inputIcon="add" />);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:138]:
```suggestion
      CypressMountWithProviders(<NumberInputComponent inputIcon="add" />);
```

File: src/components/number/number.test.js [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+
+      getElement("add").and("be.visible");
+    });
+
+    it("should check Number is disabled", () => {
+      CypressMountWithProviders(<Number disabled />);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:144]:
```suggestion
      CypressMountWithProviders(<NumberInputComponent disabled />);
```

File: src/components/number/number.test.js [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+
+      commonDataElementInputPreview().should("have.attr", "disabled");
+      commonDataElementInputPreview().should("be.disabled");
+    });
+
+    it("should render Number as read only", () => {
+      CypressMountWithProviders(<Number readOnly />);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:151]:
```suggestion
      CypressMountWithProviders(<NumberInputComponent readOnly />);
```

File: src/components/number/number.test.js [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+
+      const inputValue = "1";
+
+      commonDataElementInputPreview().type(inputValue, { force: true });
```
Comment (DlgSHi) [inline src/components/number/number.test.js:155]:
```suggestion
      commonDataElementInputPreview().realType(inputValue);
```

File: src/components/number/number.test.js [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+    });
+
+    it.each([
+      ["small", "32px"],
+      ["medium", "40px"],
+      ["large", "48px"],
+    ])(
+      "should use %s as size and render Number with %s as height",
+      (size, height) => {
+        CypressMountWithProviders(<Number size={size} />);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:168]:
```suggestion
        CypressMountWithProviders(<NumberInputComponent size={size} />);
```

File: src/components/number/number.test.js [hunk 11]
```diff
@@ -0,0 +1,329 @@
...
+    );
+
+    it("should check Number has autofocus", () => {
+      CypressMountWithProviders(<Number autoFocus />);
+
+      commonDataElementInputPreview().should("be.focused");
+    });
+
+    it("should check Number is required", () => {
+      CypressMountWithProviders(<Number label="Number" required />);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:183]:
```suggestion
      CypressMountWithProviders(<NumberInputComponent required />);
```

File: src/components/number/number.test.js [hunk 12]
```diff
@@ -0,0 +1,329 @@
...
+        const after = win.getComputedStyle($els[0], "after");
+        // read the value of the `content` CSS property
+        const contentValue = after.getPropertyValue("content");
+        // the returned value will have double quotes around it, but this is correct
+        expect(contentValue).to.eq('"*"');
+      });
+    });
+
+    it("should check Number label is inline", () => {
+      CypressMountWithProviders(<Number label="Number" labelInline />);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:198]:
```suggestion
      CypressMountWithProviders(<NumberInputComponent labelInline />);
```

File: src/components/number/number.test.js [hunk 13]
```diff
@@ -0,0 +1,329 @@
...
+    });
+
+    it.each([
+      ["right", "end"],
+      ["left", "start"],
+    ])(
+      "should use %s as labelAligment and render it with %s as css properties",
+      (alignment, cssProp) => {
+        CypressMountWithProviders(
+          <Number label="Number" labelInline labelAlign={alignment} />
```
Comment (DlgSHi) [inline src/components/number/number.test.js:212]:
```suggestion
          <NumberInputComponent labelInline labelAlign={alignment} />
```

File: src/components/number/number.test.js [hunk 14]
```diff
@@ -0,0 +1,329 @@
...
+
+    it.each([
+      ["10", "90", "135px", "1215px"],
+      ["30", "70", "405px", "945px"],
+      ["80", "20", "1080px", "270px"],
+    ])(
+      "should use %s as labelWidth, %s as inputWidth and render it with correct label and input width ratios",
+      (label, input, labelRatio, inputRatio) => {
+        CypressMountWithProviders(
+          <Number
```
Comment (DlgSHi) [inline src/components/number/number.test.js:230]:
```suggestion
          <NumberInputComponent
```

File: src/components/number/number.test.js [hunk 15]
```diff
@@ -0,0 +1,329 @@
...
+            label="Number"
```
Comment (DlgSHi) [inline src/components/number/number.test.js:231]:
```suggestion
```

File: src/components/number/number.test.js [hunk 16]
```diff
@@ -0,0 +1,329 @@
...
+    });
+
+    it.each([
+      [1000, "1"],
+      [5000, "5"],
+      [10000, "10"],
+    ])(
+      "should use %s as deferTimeout and defer onChangeDeferred event for %s seconds",
+      (timeout) => {
+        const now = Date.now();
```
Comment (DlgSHi) [inline src/components/number/number.test.js:272]:
```suggestion
```

File: src/components/number/number.test.js [hunk 17]
```diff
@@ -0,0 +1,329 @@
...
+        const callbackOnChange = cy.stub();
+        const callbackOnChangeDeff = cy.stub();
+
+        CypressMountWithProviders(
+          <NumberInputComponent
+            label="Number"
```
Comment (DlgSHi) [inline src/components/number/number.test.js:278]:
```suggestion
```

File: src/components/number/number.test.js [hunk 18]
```diff
@@ -0,0 +1,329 @@
...
+            deferTimeout={timeout}
+            onChange={callbackOnChange}
+            onChangeDeferred={callbackOnChangeDeff}
+          />
+        );
+
+        cy.clock(now);
```
Comment (DlgSHi) [inline src/components/number/number.test.js:285]:
```suggestion
        cy.clock();
```

File: src/components/number/number.test.js [hunk 19]
```diff
@@ -0,0 +1,329 @@
...
+    );
+
+    it.each([["downarrow"], ["leftarrow"], ["rightarrow"], ["uparrow"]])(
+      "should return onKeyDown event when %s key is pressed",
+      (key) => {
+        const callback = cy.stub();
+
+        CypressMountWithProviders(
+          <NumberInputComponent
+            label="Number"
```
Comment (DlgSHi) [inline src/components/number/number.test.js:312]:
```suggestion
```

File: src/components/number/number.test.js [hunk 20]
```diff
@@ -0,0 +1,329 @@
...
+            onKeyDownEnabled
+            onKeyDown={callback}
+          />
+        );
+
+        commonDataElementInputPreview()
+          .clear()
+          // .type(`{${key}}`)
```
Comment (DlgSHi) [inline src/components/number/number.test.js:320]:
```suggestion
```

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: src/components/number/number.test.js [hunk 21]
```diff
@@ -0,0 +1,329 @@
...
+    it("should check Number has autofocus", () => {
+      CypressMountWithProviders(<Number autoFocus />);
+
+      commonDataElementInputPreview().should("be.focused");
+    });
+
+    it("should check Number is required", () => {
+      CypressMountWithProviders(<Number label="Number" required />);
+
+      getDataElementByValue("label").then(($els) => {
```
[the remark this answers is not in the document]
Comment (sianford) [reply src/components/number/number.test.js]: Thanks. I created a `common-steps.js` file in the `support/component-helper` directory for this and other components to call.

**Review by DlgSHi — APPROVED** [review]
Good job, @sianford !

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**